src/sv-pipeline/scripts/format_svtk_vcf_for_gatk.py

```diff
@@ -69,162 +72,98 @@ def _parse_ploidy_table(path: Text) -> Dict[Text, Dict[Text, int]]:
-            new_genotype['CN'] = 0
-        elif new_genotype['ECN'] == 1:
-            if svtype == 'DUP':
-                new_genotype['CN'] = 1 + _cache_gt_sum(genotype['GT'])
-                new_genotype['GT'] = (None,)
-            elif _cache_gt_sum(genotype['GT']) == 0:
-                new_genotype['GT'] = (0,)
-                if svtype == 'DEL':
-                    new_genotype['CN'] = 1
-            else:
-                new_genotype['GT'] = (1,)
-                if svtype == 'DEL':
-                    new_genotype['CN'] = 0
-        else:
-            gt_sum = _cache_gt_sum(genotype['GT'])
-            if svtype == 'DUP':
-                new_genotype['CN'] = 2 + gt_sum
-                new_genotype['GT'] = (None, None)
-            elif gt_sum == 0:
-                new_genotype['GT'] = (0, 0)
-                if svtype == 'DEL':
-                    new_genotype['CN'] = 2
-            elif gt_sum == 1:
-                new_genotype['GT'] = (0, 1)
-                if svtype == 'DEL':
-                    new_genotype['CN'] = 1
+        record.ref = 'N'
+    if svtype == 'BND':
+        record.info['END2'] = bnd_end_dict[record.id] if bnd_end_dict is not None \
+            else record.info.get('END2', record.stop)
+    # Fix this weird edge case (may be from CPX review workflow)
+    if svtype == 'INV' and '<CPX>' in record.alleles[1]:
+        svtype = 'CPX'
+        record.info['SVTYPE'] = svtype
+    is_ddup = svtype == 'CPX' and 'dDUP' in record.info.get('CPX_TYPE', '')
+    if svtype == 'BND' or svtype == 'INS' or svtype == 'CTX' or is_ddup:
+        record.stop = record.start + 1
+        if is_ddup:
+            # e.g. SOURCE=DUP_chrX:49151588-49151850
+            source = record.info.get('SOURCE', None)
+            if source is not None:
+                tokens = source.split(':')
+                chr2 = tokens[0].split('_')[-1]
+                end2 = int(tokens[-1].split('-')[0])
+                record.info['CHR2'] = chr2
+                record.info['END2'] = end2
             else:
-                new_genotype['GT'] = (1, 1)
-                if svtype == 'DEL':
-                    new_genotype['CN'] = 0
-        if svtype == 'CNV':
-            new_genotype['GT'] = (None,) * new_genotype['ECN']
-    return new_record
+                # Sometimes SOURCE is not set (may be from CPX review workflow)
+                record.info['CHR2'] = record.chrom
+                record.info['END2'] = record.stop
+    # Delete empty INFO fields (GATK does not like "." for non-String types)
+    keys = record.info.keys()
+    for k in keys:
+        val = record.info[k]
+        if is_null(val) or (isinstance(val, tuple) and len(val) == 1 and is_null(val[0])):
+            del record.info[k]
+    # copy FORMAT fields
+    for sample, genotype in record.samples.items():
+        genotype['ECN'] = ploidy_dict[sample][contig]
+    if scale_down_gq:
+        rescale_gq(record)
+    return record
 
 
 def _cache_gt_sum(gt):
```

Comment:
This function doesn't appear to be used anymore in this script

Reply:
Thanks deleted this and the one below it too